src/js/app.js

```diff
@@ -0,0 +1,196 @@
+"use strict";
+
+angular.module('invertedIndex', [])
+  .filter('range', () => {
+    return function (input, number) {
+      for (let i = 1; parseInt(number) >= i; i++) {
+        input.push(i);
+      }
+      return input;
+    };
+  })
+  .directive('fileUpload', ['alerts', (alerts) => {
+    /*
+     * template to be rendered
+     */
+    const template = `<input type="file" id="file" name="files" multiple>
+                      <label for="file">
+                      <i class="fa fa-upload" aria-hidden="true"></i> 
+                      Upload A file <span>{{ uploadedFilesCount }}
+                      </span></label>`;
+
+
+    /*
+     * link funtion to be called
+     */
+    function link(scope, elem) {
+      elem.on('change', (evt) => {
+        const fileList = evt.target.files;
+        angular.forEach(fileList, (file) => {
+          if (!scope.isValidFile(file.name)) {
+            scope.$apply(function () {
+              alerts("please all file must be a valid json",
+                "danger", true, 5000);
+            });
+            return;
+          }
+          const reader = new FileReader();
+          //event fired when reader.readAsTex is called
+          reader.onload = function (event) {
+            try {
+              const result = JSON.parse(event.target.result);
+              scope.$apply(() => {
+                scope.uploadedFiles[file.name] = result;
+              });
+            } catch (e) {
+              scope.$apply(() => {
+                alerts("invalid json. pls refer to index guide ",
+                  "danger", true, 5000);
+              });
+              return;
+            }
+          };
+          reader.readAsText(file);
+        });
+      });
+    }
+    return {
+      template: template,
+      link: link
+    };
+  }])
+  .controller('mainController', ['$scope', 'alerts', ($scope, alerts) => {
+    /**
+     *  {Object} instanciates the invertedIndex class
+     */
+    const indexer = new InvertedIndex();
+
+    /**
+     *  {Object}  uploaded files
+     */
+    $scope.uploadedFiles = {};
+
+    /** 
+     *  {Object} indexed value to display
```

Comment:
**Nitpick**
You can remove unnecessary comments. The variable names are self-explanatory enough